documentation/preview-push.md

```diff
@@ -32,71 +32,80 @@ OPTIONS:
+  * Specifies the name of the credentials file, normally `creds.json`.
+    Typically the file is read.  If the executable bit is set, the file is
+    executed and the output is used as the configuration. See
+    [creds.json][creds-json.md] for details.
+
+* `--providers name,name2`
+  * Specifies a comma-separated list of providers to
+    enable.  The default is all providers.  A provider can opt out of being in the
+    default list by `"_exclude_from_defaults": "true"` to the credentials entry for
+    that provider. In that case, the provider will only be activated if it is
+    included in `--providers`.
+
+* `--domains value`
+  * Specifies a comma-separated list of domains to include.
+    Typically all domains are included in `preview`/`push`.  Wildcards are not
+    permitted except `*` at the start of the entry. For example, `--domains
+    example.com,*.in-addr.arpa` would include `example.com` plus all reverse lookup
+    domains.
+
+* `--v foo=bar`
+  * Sets the variable `foo` to the value `bar` prior to
+    interpreting the configuration file.  Multiple `-v` options can be used.
+
+* `--notify`
+  * Enables sending notifications to the destinations configured in `creds.json`.
+
+* `--dev`
+  * Developer mode.  Normally `helpers.js` is embedded in the dnscontrol
+    executable.  With this flag, the local file `helpers.js` is read instead.
+
+* `--expect-no-changes`
+  * If set, a non-zero exit code is returned if there are
+    changes.  Normally DNSControl sets the exit code based on whether or not there
+    were protocol errors or other reasons the program can not continue. With this
+    flag set, the exit code indicates if any changes were required.  This is
+    typically used with `preview` to allow scripts to determine if changes would
+    happen if `push` was used. For example, one might want to run `dnscontrol
+    preview --expect-no-changes` daily to determine if changes have been made to
+    a domain outside of DNSControl.
+
+* `--no-populate`
+  * Do not auto-create non-existing zones at the provider.
+    Normally non-existent zones are automatically created at a provider (unless the
+    provider does not implement zone creation). This flag disables that feature.
+
+* `--full`
+  * Add headings, providers names, notifications of no changes, etc. to
+    the output.  Normally the output of `preview`/`push` is extremely brief. This
+    makes the output more verbose. Useful for debugging.
+
+* `--bindserial value`
+  * Force BIND serial numbers to this value. Normally the
+    BIND provider generates SOA serial numbers automatically. This flag forces the
+    serial number generator to output the value specified for all domains.  This is
+    generally used for reproducibility in testing pipelines.
+
+* `--report name`
+  * (`push` only!)  Generate a machine-parseable report of
+    performed corrections in the file named `name`.  If no name is specified, no
+    report is generated.
+
+## Starting in v4.9
```

Comment:
Perhaps it is better to stay in line with the other H1 title. Then to say that these commands exist from 4.9.

```suggestion
## ppreview/ppush

{% hint style="info" %}
Starting in v4.9
{% endhint %}
```

Reply:
Fixed by https://github.com/StackExchange/dnscontrol/pull/2891